test/regfox/regfox_api.test.js

```diff
@@ -250,4 +257,43 @@ describe('regfox_api', () => {
       return addNote(undefined, 'fake fake', 1234).should.eventually.be.rejected;
     });
   });
+
+  describe('checkIn', () => {
+    const ID = 12345312;
+
+    const getResponse = stub();
+    const server = setupServer(...checkInHandler(ID, getResponse));
+    before(() => server.listen());
+    afterEach(() => getResponse.reset());
+    afterEach(() => server.resetHandlers());
+    after(() => server.close());
+
+    it('returns data as requested', async () => {
+      getResponse.returns();
+
+      const result = await checkIn(ID, 1234);
+
+      expect(result).to.deep.equal({ 'status': 'OK' });
+    });
+
+    it('handles an illegal request', async () => {
+      getResponse.returns({
+        code: 1000,
+        message: 'bad request',
+      });
+
+      return checkIn(ID, 1234).should.eventually.be.rejected;
+    });
+
+    it('handles a duplicate request', async () => {
+      getResponse.returns({
+        code: 1000,
```

Comment:
Is it really the same code for both?

Reply:
Yes. that surprised me at first too. You can see the commit [here](https://github.com/anthroarts/RegExtension/commit/5e9992011e855b019ebe5db60ccf0d3bb25f0ad1#diff-a4e9c7b0222ec040a3618347a2b511d482a5dc812c6f66b21f924e7caec8c4f1R174) where I discovered "Oh no, they use code 1000 to mean everything."